backend/src/test/java/com/pickpick/message/application/ReminderServiceTest.java

```diff
@@ -61,293 +67,371 @@ class ReminderServiceTest {
+                    .collect(Collectors.toList());
+
+            for (Message fixture : messagesInChannel) {
+                messages.save(fixture);
+            }
+
+            return messagesInChannel;
+        }
+
+        private List<Reminder> saveRemindersAtDifferentTimes(final Member member, final List<Message> messages) {
+            List<Reminder> savedReminders = new ArrayList<>();
+            int totalSize = messages.size();
+
+            for (int i = 0; i < totalSize; i++) {
+                LocalDateTime remindDateTime = LocalDateTime.now().plusHours(i + 1);
+                Reminder reminder = reminders.save(
+                        new Reminder(member, messages.get(totalSize - i - 1), remindDateTime));
+                savedReminders.add(reminder);
+            }
+
+            return savedReminders;
+        }
+
+        private Message saveDummyMessage(final Member member, final Channel channel) {
+            Message message = new Message(UUID.randomUUID().toString(), "리마인더 설정 용 메시지", member, channel,
+                    LocalDateTime.now(), LocalDateTime.now());
+            return messages.save(message);
+        }
+
+        private List<Long> extractIds(final List<Reminder> reminders) {
+            return reminders.stream()
+                    .map(Reminder::getId)
+                    .collect(Collectors.toList());
+        }
+
+        private List<Long> extractOrderedByRemindDateIds(final List<Reminder> reminders) {
+            return reminders.stream()
+                    .sorted(Comparator.comparing(Reminder::getRemindDate))
+                    .map(Reminder::getId)
+                    .collect(Collectors.toList());
+        }
+
+        private List<Long> extractOrderedByRemindDateIdsDefaultLimit(final List<Reminder> reminders) {
+            return reminders.stream()
+                    .sorted(Comparator.comparing(Reminder::getRemindDate))
+                    .map(Reminder::getId)
+                    .limit(20)
+                    .collect(Collectors.toList());
+        }
+
+        private List<Long> extractFutureFromTargetOrderedByRemindDateIds(final List<Reminder> reminders,
+                                                                         final Reminder target, final int limit) {
+            return reminders.stream()
+                    .filter(reminder -> reminder.getRemindDate().isAfter(target.getRemindDate()))
+                    .sorted(Comparator.comparing(Reminder::getRemindDate))
+                    .map(Reminder::getId)
+                    .limit(limit)
+                    .collect(Collectors.toList());
+        }
     }
 
-    @DisplayName("다른 사용자의 리마인더 삭제시 예외")
-    @Test
-    void deleteOtherMembers() {
-        // given & when & then
-        assertThatThrownBy(() -> reminderService.delete(1L, 1L))
-                .isInstanceOf(ReminderDeleteFailureException.class);
+    @DisplayName("리마인더 저장/삭제/수정/단건 조회 시")
```

Comment:
```suggestion
    @DisplayName("리마인더 변경 시") 
    // @DisplayName("리마인더 저장/삭제/수정 시") 
    // @DisplayName("리마인더 CUD 시") 
```
요 부분은 CUD 관련만 하는게 어떨까요? 

윗쪽의 `@DisplayName("복수의 리마인더 조회 시")`를 `@DisplayName("리마인더 조회 시")`로 변경한 뒤에 해당 부분에서 조회 관련 모든 테스트를 진행 하는것도 좋을 것 같아요😄

Reply:
이렇게 하지 않았던 이유가, 단건 조회를 테스트 하려면 제일 외부 `@Nested` 클래스 필드로 선언된 픽스쳐와 별도로 단건 조회 테스트 내에서 단건 픽스쳐를 저장해야 한다고 생각했기 때문인데요
그럴 경우에 테스트 실행 순서에 따라 다른 테스트가 영향을 받을 수도 있고, 정해진 픽스쳐에서 어떻게 조회되는가를 테스트한다는 `@Nested` 도입 취지에 어긋나기도 해서 이렇게 짰었습니다 
그런데 저도 조회가 복수/단건 나뉘어져 있는 게 마음에 안들어서... 
- 단건 조회는 저장된 리마인더에서 messageId, memberId를 꺼내서 한다 
- 단건 조회 예외 발생은, 공유 픽스쳐와 무관한 새로운 멤버를 저장해서 다른 테스트에 영향을 끼칠 일이 없도록 한다  

이 방향으로 수정해 단건도 같은 `Nested` 안으로 넣었습니다 😉